### PR TITLE
Grooming (compilation warning and potential issue)

### PR DESCRIPTION
**Pull request by sarroutbi**
Signed-off-by: Sergio Arroutbi <sarroutb@redhat.com>

### CODE REVIEW

File: src/rdkafka_broker.c [hunk 1]
```diff
@@ -76,6 +76,7 @@
 
 
 static const int rd_kafka_max_block_ms = 1000;
+static const unsigned short HOLBSTR_MAX_SIZE = 256;
```
Comment (edenhill) [inline src/rdkafka_broker.c:79]: Why is this change needed?
That buffer size is local to the timeout_scan() function and only used once, so no need for a predefined value.

File: src/rdkafka_idempotence.c [hunk 2]
```diff
@@ -313,14 +313,16 @@ void rd_kafka_idemp_pid_fsm(rd_kafka_t *rk) {
...
-                rd_kafka_broker_destroy(rkb);
-
                 if (err) {
                         rd_rkb_dbg(rkb, EOS, "GETPID",
                                    "Can't acquire ProducerId from "
                                    "this broker: %s",
                                    errstr);
+                }
 
+                rd_kafka_broker_destroy(rkb);
```
Comment (edenhill) [inline src/rdkafka_idempotence.c:323]: Good find :+1:

File: src/rdkafka_mock.c [hunk 3]
```diff
@@ -1765,6 +1765,7 @@ void rd_kafka_mock_push_request_errors(rd_kafka_mock_cluster_t *mcluster,
                 errors[i] = va_arg(ap, rd_kafka_resp_err_t);
 
         rd_kafka_mock_push_request_errors_array(mcluster, ApiKey, cnt, errors);
+        va_end(ap);
```
Comment (edenhill) [inline src/rdkafka_mock.c:1768]: nit: Move this to the line after the for-loop

**Review by edenhill — CHANGES_REQUESTED** [review]

File: src/rdkafka_broker.c (hunk 1 above)
Comment (sarroutbi) [reply src/rdkafka_broker.c:79]: The important thing here is the change in the value to avoid compilation warning. Regarding the usage of a constant, I just prefer using constants that define what is a buffer for (to avoid magic numbers). However, if not necessary, I can return it to where it was.

Comment (edenhill) [reply src/rdkafka_broker.c:79]: Ah, I see. 
Yes, please remove the constant and just the use the value directly.

Comment (sarroutbi) [reply src/rdkafka_broker.c]: Sure. Done.

File: src/rdkafka_mock.c (hunk 3 above)
Comment (sarroutbi) [reply src/rdkafka_mock.c]: Done

**Comment by edenhill** [comment]
Thank you!

I notice that your branch is called covscan.. were these changes made as a result of a Coverity scan? If so; are there more to come?

I did a coverity scan a couple of years ago, which perhaps found one or two issues, but had hordes of false positives. Maybe the situation has improved?

**Comment by sarroutbi** [comment]
Hello @edenhill. You are welcome.

Yes, this changes were result of a covscan execution in RHEL version of librdkafka. As you say, there were many issues, and the majority of them were false positives, but some of them were also "true positives". I analyzed all them and ended up with this PR, so from my side I expect no more changes related to covscan in the short trem.
